streaming/src/main/scala/org/apache/gearpump/streaming/StreamApplication.scala

```diff
@@ -77,11 +77,25 @@ class StreamApplication(override val name : String,  inputUserConfig: UserConfig
 object StreamApplication {
 
   def apply (name : String, dag: Graph[_ <: Processor[Task], _ <: Partitioner], userConfig: UserConfig): StreamApplication = {
-    val graph = dag.mapVertex(Processor.ProcessorToProcessorDescription(_))
+    import Processor._
+    val graph = dag.mapVertex(ProcessorToProcessorDescription)
     new StreamApplication(name, userConfig, graph)
   }
 
+  def apply (name : String, jarName: String, processors: Map[ProcessorId, ProcessorDescription], dag: Graph[Int, String]): StreamApplication = {
```

Comment:
In that case, how about 

```
def apply (name : String, dag: Graph[String, String], userConfig: UserConfig): StreamApplication = {
}
```

The first parameter of Graph is the processor, then second is partitioner.